src/components/QrScan/QrScan.js

```diff
@@ -1,56 +1,64 @@
-import React, { useState } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import { Button, Grid } from '@mui/material';
 import { makeStyles } from '@mui/styles';
-import QrReader from 'react-qr-reader';
 import { useHistory } from 'react-router-dom';
 import { v4 as uuidv4 } from 'uuid';
 import frame from 'assets/frame.png';
 import { useQrDataContext } from 'components/QrDataProvider';
+import QrScanner from './vendor/qr-scanner';
+
+QrScanner.WORKER_PATH = '/shc-web-verifier/qr-scanner-worker.min.js';
 
 const useStyles = makeStyles(() => ({
   button: {
     '&:hover': {
-      cursor: 'default'
-    }
+      cursor: 'default',
+    },
   },
   frame: {
     position: 'relative',
     height: '550px',
     width: '640px',
     marginTop: '3rem',
     marginBottom: '3rem',
-    zIndex: '2',
+    zIndex: '3',
   },
   qrScanner: {
+    width: '570px',
+    height: '500px',
+    'object-fit': 'cover',
     position: 'absolute',
     marginLeft: '2.2rem',
     marginTop: '4.5rem',
-    height: '500px',
-    width: '570px',
-    zIndex: '1',
+    zIndex: '2',
     '& section': {
       position: 'unset !important',
       '& div': {
-        boxShadow: 'unset !important'
+        boxShadow: 'unset !important',
       },
     },
-  }
+  },
 }));
 
 const healthCardPattern = /^shc:\/(?<multipleChunks>(?<chunkIndex>[0-9]+)\/(?<chunkCount>[0-9]+)\/)?[0-9]+$/;
 
+let qrScan;
+
 const QrScan = () => {
   const history = useHistory();
   const classes = useStyles();
   const { setQrCodes } = useQrDataContext();
   const [scannedCodes, setScannedCodes] = useState([]);
+  const [scannedData, setScannedData] = useState('');
+  const runningQrScanner = useRef(null);
 
   const handleScan = (data) => {
```

Comment:
In React, using a function as a dependency inside a useEffect hook can cause a warning, because the function will be recreated on each render, meaning it will point to a different place in memory and force the useEffect hook to continuously re-render.

I didn't really know too much about this, but there is some good documentation here that explains the issue pretty well: https://typeofnan.dev/fix-function-makes-the-dependencies-of-useEffect-hook-change-on-every-render-warning-in-react/.

There are two ways to fix this. The easiest way to fix this for the HandleScan function is probably to put it inside UseEffect hook, since it is only used there. Alternatively you can also try to use the same method as described in the comment for `handleError`.

Moving handleScan into the second useEffect hook like this eliminates the warning:
```js
  // Call handleScan() from parent component when the QrScanner successfully reads data
  useEffect(() => {
    const handleScan = (data) => {
      if (healthCardPattern.test(data)) {
        const match = data.match(healthCardPattern);
        if (match.groups.multipleChunks) {
          const chunkCount = +match.groups.chunkCount;
          const currentChunkIndex = +match.groups.chunkIndex;
          let tempScannedCodes = [...scannedCodes];
          if (tempScannedCodes.length !== chunkCount) {
            tempScannedCodes = new Array(chunkCount);
            tempScannedCodes.fill(null, 0, chunkCount);
          }

          if (tempScannedCodes[currentChunkIndex - 1] === null) {
            tempScannedCodes[currentChunkIndex - 1] = data;
          }

          if (tempScannedCodes.every((code) => code)) {
            setQrCodes(tempScannedCodes);
            history.push('/display-results');
          }
          setScannedCodes(tempScannedCodes);
        } else {
          setQrCodes([data]);
          history.push('/display-results');
        }
      }
    };

    if (scannedData) {
      try {
        handleScan(scannedData);
      } catch (e) {
        handleError();
      }
    }
  }, [scannedData, handleError, history, scannedCodes, setQrCodes]);
```

Reply:
Thank you for providing that documentation! I had a hard time understanding most of the documentation online regarding useEffect and callback issues, so I found this very helpful. Making this change got rid of the warning!